kythe/go/platform/kcd/kythe/units.go

```diff
@@ -94,36 +93,12 @@ func (u Unit) Canonicalize() {
 }
 
 // Digest satisfies part of the kcd.Unit interface.
-func (u Unit) Digest(w io.Writer) {
+func (u Unit) Digest(w io.Writer) error {
 	pb := u.Proto
 	if pb == nil {
 		pb = new(apb.CompilationUnit)
 	}
-	put := func(tag string, ss ...string) {
```

Comment:
This algorithm is likely a better choice than just the JSON encoded proto (but we should work on documenting it and implementing it in the other languages).

Reply:
Especially since we don't have any control over the JSON representation emitted, which could change at any time.

Reply:
Ok. I was just going off what was in the kzip doc, which explicitly states a JSON encoding. What is currently in units.go is definitely not JSON encoding.